nano/test_common/testutil.hpp

```diff
@@ -254,6 +266,28 @@ namespace test
 		T & ref;
 	};
 
+	/**
+	 * A container that constructs and stores an object and calls `start` from constructor and `stop` from destructor (RAII)
+	 */
+	template <typename T>
+		requires start_stoppable<T>
+	struct start_stop_container
+	{
+		template <typename... Args>
+		explicit start_stop_container (Args &&... args) :
```

Comment:
I'm having trouble understanding why this class is implement this way, this doesn't seem to be in-line with the goal of having a variadic start stop guard.

Reply:
I am trying to create a class that will hold an object and call start and stop on it.
Is there a bug or are against the idea?